wagtail_newsletter/templates/wagtail_newsletter/panels/newsletter_panel.html

```diff
@@ -1,31 +1,101 @@
 {% load wagtailadmin_tags %}
 
-<div class="wn-panel">
+<div
+    class="wn-panel"
+    data-controller="wn-panel"
+    data-wn-panel-user-email-value="{{ user_email }}"
+    data-action="wn-test:send@window->wn-panel#test"
+>
     {% if error_message %}
         <div class="help-block help-critical">
             {% icon name="warning" %}
             {{ error_message }}
         </div>
     {% endif %}
 
-    <div class="help-block help-info">
-        {% icon name="help" %}
-        This action will save a new page revision.
-    </div>
-
-    <p>
-        <button
-            type="submit"
-            class="button button-secondary button-longrunning"
-            name="newsletter_action"
-            value="save_campaign"
-            data-controller="w-progress"
-            data-action="w-progress#activate"
-        >
-            {% icon name="spinner" %}
-            Save campaign to {{ backend_name }}
-        </button>
-    </p>
+    {% if not campaign_sent %}
+        <div class="help-block help-info">
+            {% icon name="help" %}
+            These actions will save a new page revision.
+        </div>
+
+        <p>
+            <input
+                type="hidden"
+                name="newsletter_test_email_address"
+                data-wn-panel-target="testAddress"
+            >
+
+            <button
+                type="submit"
+                class="button button-secondary button-longrunning"
+                name="newsletter_action"
+                value="save_campaign"
+                data-controller="w-progress"
+                data-action="w-progress#activate"
+            >
+                {% icon name="spinner" %}
+                Save campaign to {{ backend_name }}
+            </button>
+
+            <button
+                type="button"
+                class="button button-secondary button-longrunning"
+                data-a11y-dialog-show="wn-test-dialog"
+                data-controller="w-progress"
+                data-wn-panel-target="testButton"
+            >
+                {% icon name="spinner" %}
+                Send test email
+            </button>
+
+            {% dialog icon_name="mail" id="wn-test-dialog" title="Send test email" %}
+                <form
+                    data-controller="wn-test"
+                    data-action="wn-test#send:prevent"
+                >
```

Comment:
The contents of the dialog needs a separate Stimulus controller because it lives outside of the page edit form.